### ISSUE TITLE
Issue of Jar dependency with Jruby-9.3.3.0

### ISSUE THREAD

**Issue by Kinju011** [jruby/jruby#7059]
**Summary**
Hello
I am working on Neo4j-Ruby driver where I am facing an issue while installing dependencies

I have checked this with [Jruby-9.3.2.0, Jruby-9.3.1.0] where it is working fine, below I have provided details of the error
here is a [link](https://github.com/neo4jrb/neo4j-ruby-driver/runs/5001004372?check_suite_focus=true) where one can check the details.

**Environment Information**
  
* Jruby version - 9.3.3.0

**Expected Behavior**

* Jar file dependency should be fixed

**Actual Behavior**

Using neo4j-ruby-driver 4.4.0.20220131034407 (java) from source at `.`
  jar dependencies for neo4j-ruby-driver-4.4.0.20220131034407-java.gemspec . . .
Error: ] Some problems were encountered while processing the POMs:
[FATAL] Non-parseable POM /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/gemspec_pom.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:1)  @ line 1, column 1
 @ 
Error:  The build could not read 1 project -> [Help 1]
Error:    
Error:    The project  (/home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/gemspec_pom.rb) has 1 error
Error:      Non-parseable POM /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/gemspec_pom.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:1)  @ line 1, column 1 -> [Help 2]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
Error:  [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/ModelParseException
      org.neo4j.driver:neo4j-java-driver:4.4.2:compile

```
Errno::ENOENT: No such file or directory - /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/psych-4.0.3-java/deps.lst
  org/jruby/RubyIO.java:1227:in `sysopen'
  org/jruby/RubyIO.java:3986:in `read'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/installer.rb:69:in `load_from_maven'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/installer.rb:235:in `install_dependencies'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/installer.rb:224:in `do_install'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/installer.rb:170:in `vendor_jars'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/post_install_hook.rb:28:in `block in <main>'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/rubygems/installer.rb:367:in `block in run_post_install_hooks'
  org/jruby/RubyArray.java:1865:in `each'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/rubygems/installer.rb:366:in `run_post_install_hooks'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/rubygems_gem_installer.rb:43:in `install'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/source/rubygems.rb:204:in `install'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/installer/gem_installer.rb:54:in `install'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/installer/parallel_installer.rb:186:in `do_install'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/installer/parallel_installer.rb:177:in `block in worker_pool'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/worker.rb:62:in `apply_func'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/worker.rb:57:in `block in process_queue'
  org/jruby/RubyKernel.java:1507:in `loop'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/worker.rb:54:in `process_queue'
  /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared/gems/bundler-2.3.6/lib/bundler/worker.rb:91:in `block in create_threads'
```

## Environment

```
Bundler       2.3.6
  Platforms   ruby, universal-java-11
Ruby          2.6.8p0 (2022-01-19 revision 67951) [universal-java-11]
  Full Path   /home/runner/.rubies/jruby-9.3.3.0/bin/jruby
  Config Dir  /home/runner/.rubies/jruby-9.3.3.0/etc
RubyGems      3.2.29
  Gem Home    /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared
  Gem Path    /home/runner/.local/share/gem/jruby/2.6.0:/home/runner/.rubies/jruby-9.3.3.0/lib/ruby/gems/shared
  User Home   /home/runner
  User Path   /home/runner/.local/share/gem/jruby/2.6.0
  Bin Dir     /home/runner/.rubies/jruby-9.3.3.0/bin
OpenSSL       
  Compiled    JRuby-OpenSSL 0.11.0
  Loaded      JRuby-OpenSSL 0.11.0
  Cert File   /opt/hostedtoolcache/jdk/11.0.14/x64/lib/security/cacerts
  Cert Dir    /etc/ssl/certs
Tools         
  Git         2.34.1
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2022-01-26
Git SHA           056f64c33a
Released Version  true
```

### Gemfile

```ruby
# -*- ruby -*-

# DO NOT EDIT THIS FILE. Instead, edit Rakefile, and run `rake bundler:gemfile`.

source "https://rubygems.org/"
gemspec

# vim: syntax=ruby
```

### Gemfile.lock

```
<No /home/runner/work/neo4j-ruby-driver/neo4j-ruby-driver/Gemfile.lock found>
```

### neo4j-ruby-driver.gemspec

```ruby
# -*- encoding: utf-8 -*-
# stub: neo4j-ruby-driver 4.4.0.20220131034407 java libruby

Gem::Specification.new do |s|
  s.name = "neo4j-ruby-driver".freeze
  s.version = "4.4.0.20220131034407"
  s.platform = "java".freeze

  s.required_rubygems_version = Gem::Requirement.new(">= 0".freeze) if s.respond_to? :required_rubygems_version=
  s.metadata = { "homepage_uri" => "https://github.com/neo4jrb/neo4j-ruby-driver" } if s.respond_to? :metadata=
  s.require_paths = ["lib".freeze, "ruby".freeze]
  s.authors = ["Heinrich Klobuczek".freeze]
  s.date = "2022-01-31"
  s.description = "".freeze
  s.email = ["heinrich@mail.com".freeze]
  s.extra_rdoc_files = ["LICENSE.txt".freeze, "README.md".freeze]
  s.files = ["LICENSE.txt".freeze, "README.md".freeze, "lib/loader.rb".freeze, "lib/neo4j/driver/auto_closable.rb".freeze, "lib/neo4j/driver/exceptions/authentication_exception.rb".freeze, "lib/neo4j/driver/exceptions/authorization_expired_exception.rb".freeze, "lib/neo4j/driver/exceptions/certificate_exception.rb".freeze, "lib/neo4j/driver/exceptions/client_exception.rb".freeze, "lib/neo4j/driver/exceptions/connection_read_timeout_exception.rb".freeze, "lib/neo4j/driver/exceptions/database_exception.rb".freeze, "lib/neo4j/driver/exceptions/discovery_exception.rb".freeze, "lib/neo4j/driver/exceptions/fatal_discovery_exception.rb".freeze, "lib/neo4j/driver/exceptions/illegal_state_exception.rb".freeze, "lib/neo4j/driver/exceptions/neo4j_exception.rb".freeze, "lib/neo4j/driver/exceptions/no_such_record_exception.rb".freeze, "lib/neo4j/driver/exceptions/protocol_exception.rb".freeze, "lib/neo4j/driver/exceptions/result_consumed_exception.rb".freeze, "lib/neo4j/driver/exceptions/security_exception.rb".freeze, "lib/neo4j/driver/exceptions/service_unavailable_exception.rb".freeze, "lib/neo4j/driver/exceptions/session_expired_exception.rb".freeze, "lib/neo4j/driver/exceptions/token_expired_exception.rb".freeze, "lib/neo4j/driver/exceptions/transaction_nesting_exception.rb".freeze, "lib/neo4j/driver/exceptions/transient_exception.rb".freeze, "lib/neo4j/driver/exceptions/untrusted_server_exception.rb".freeze, "lib/neo4j/driver/exceptions/value/lossy_coercion.rb".freeze, "lib/neo4j/driver/exceptions/value/not_multi_valued.rb".freeze, "lib/neo4j/driver/exceptions/value/uncoercible.rb".freeze, "lib/neo4j/driver/exceptions/value/unsizable.rb".freeze, "lib/neo4j/driver/exceptions/value/value_exception.rb".freeze, "lib/neo4j/driver/internal/duration_normalizer.rb".freeze, "lib/neo4j/driver/internal/validator.rb".freeze, "lib/neo4j/driver/types/bytes.rb".freeze, "lib/neo4j/driver/types/local_date_time.rb".freeze, "lib/neo4j/driver/types/local_time.rb".freeze, "lib/neo4j/driver/types/offset_time.rb".freeze, "lib/neo4j/driver/types/point.rb".freeze, "lib/neo4j/driver/types/time.rb".freeze, "lib/neo4j/driver/version.rb".freeze, "lib/neo4j_ruby_driver.rb".freeze, "ruby/neo4j/driver.rb".freeze, "ruby/neo4j/driver/auth_tokens.rb".freeze, "ruby/neo4j/driver/config.rb".freeze, "ruby/neo4j/driver/graph_database.rb".freeze, "ruby/neo4j/driver/internal/async/connection/bolt_protocol_util.rb".freeze, "ruby/neo4j/driver/internal/async/connection/bootstrap_factory.rb".freeze, "ruby/neo4j/driver/internal/async/connection/channel_attributes.rb".freeze, "ruby/neo4j/driver/internal/async/connection/channel_connected_listener.rb".freeze, "ruby/neo4j/driver/internal/async/connection/channel_connector_impl.rb".freeze, "ruby/neo4j/driver/internal/async/connection/channel_pipeline_builder_impl.rb".freeze, "ruby/neo4j/driver/internal/async/connection/direct_connection.rb".freeze, "ruby/neo4j/driver/internal/async/connection/event_loop_group_factory.rb".freeze, "ruby/neo4j/driver/internal/async/connection/handshake_completed_listener.rb".freeze, "ruby/neo4j/driver/internal/async/connection/handshake_handler.rb".freeze, "ruby/neo4j/driver/internal/async/connection/netty_channel_initializer.rb".freeze, "ruby/neo4j/driver/internal/async/connection/netty_domain_name_resolver.rb".freeze, "ruby/neo4j/driver/internal/async/connection/netty_domain_name_resolver_group.rb".freeze, "ruby/neo4j/driver/internal/async/connection/routing_connection.rb".freeze, "ruby/neo4j/driver/internal/async/immutable_connection_context.rb".freeze, "ruby/neo4j/driver/internal/async/inbound/byte_buf_input.rb".freeze, "ruby/neo4j/driver/internal/async/inbound/channel_error_handler.rb".freeze, "ruby/neo4j/driver/internal/async/inbound/chunk_decoder.rb".freeze, "ruby/neo4j/driver/internal/async/inbound/connect_timeout_handler.rb".freeze, "ruby/neo4j/driver/internal/async/inbound/connection_read_timeout_handler.rb".freeze, "ruby/neo4j/driver/internal/async/inbound/inbound_message_dispatcher.rb".freeze, "ruby/neo4j/driver/internal/async/inbound/inbound_message_handler.rb".freeze, "ruby/neo4j/driver/internal/async/inbound/message_decoder.rb".freeze, "ruby/neo4j/driver/internal/async/internal_async_session.rb".freeze, "ruby/neo4j/driver/internal/async/internal_async_transaction.rb".freeze, "ruby/neo4j/driver/internal/async/leak_logging_network_session.rb".freeze, "ruby/neo4j/driver/internal/async/network_connection.rb".freeze, "ruby/neo4j/driver/internal/async/network_session.rb".freeze, "ruby/neo4j/driver/internal/async/outbound/chunk_aware_byte_buf_output.rb".freeze, "ruby/neo4j/driver/internal/async/outbound/outbound_message_handler.rb".freeze, "ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb".freeze, "ruby/neo4j/driver/internal/async/pool/netty_channel_health_checker.rb".freeze, "ruby/neo4j/driver/internal/async/pool/netty_channel_pool.rb".freeze, "ruby/neo4j/driver/internal/async/pool/netty_channel_tracker.rb".freeze, "ruby/neo4j/driver/internal/async/pool/network_connection_factory.rb".freeze, "ruby/neo4j/driver/internal/async/pool/pool_settings.rb".freeze, "ruby/neo4j/driver/internal/async/result_cursors_holder.rb".freeze, "ruby/neo4j/driver/internal/async/unmanaged_transaction.rb".freeze, "ruby/neo4j/driver/internal/bolt_server_address1.rb".freeze, "ruby/neo4j/driver/internal/cluster/cluster_composition.rb".freeze, "ruby/neo4j/driver/internal/cluster/cluster_composition_lookup_result.rb".freeze, "ruby/neo4j/driver/internal/cluster/cluster_routing_table.rb".freeze, "ruby/neo4j/driver/internal/cluster/identity_resolver.rb".freeze, "ruby/neo4j/driver/internal/cluster/loadbalancing/least_connected_load_balancing_strategy.rb".freeze, "ruby/neo4j/driver/internal/cluster/loadbalancing/load_balancer.rb".freeze, "ruby/neo4j/driver/internal/cluster/loadbalancing/round_robin_array_index.rb".freeze, "ruby/neo4j/driver/internal/cluster/multi_databases_routing_procedure_runner.rb".freeze, "ruby/neo4j/driver/internal/cluster/rediscovery_impl.rb".freeze, "ruby/neo4j/driver/internal/cluster/route_message_routing_procedure_runner.rb".freeze, "ruby/neo4j/driver/internal/cluster/routing_context.rb".freeze, "ruby/neo4j/driver/internal/cluster/routing_procedure_cluster_composition_provider.rb".freeze, "ruby/neo4j/driver/internal/cluster/routing_procedure_response.rb".freeze, "ruby/neo4j/driver/internal/cluster/routing_settings.rb".freeze, "ruby/neo4j/driver/internal/cluster/routing_table_handler_impl.rb".freeze, "ruby/neo4j/driver/internal/cluster/routing_table_registry_impl.rb".freeze, "ruby/neo4j/driver/internal/cluster/single_database_routing_procedure_runner.rb".freeze, "ruby/neo4j/driver/internal/connection_settings.rb".freeze, "ruby/neo4j/driver/internal/cursor/async_result_cursor_impl.rb".freeze, "ruby/neo4j/driver/internal/cursor/async_result_cursor_only_factory.rb".freeze, "ruby/neo4j/driver/internal/cursor/disposable_async_result_cursor.rb".freeze, "ruby/neo4j/driver/internal/cursor/result_cursor_factory_impl.rb".freeze, "ruby/neo4j/driver/internal/cursor/rx_result_cursor_impl.rb".freeze, "ruby/neo4j/driver/internal/database_name.rb".freeze, "ruby/neo4j/driver/internal/database_name_util.rb".freeze, "ruby/neo4j/driver/internal/default_bookmark_holder.rb".freeze, "ruby/neo4j/driver/internal/default_domain_name_resolver.rb".freeze, "ruby/neo4j/driver/internal/direct_connection_provider.rb".freeze, "ruby/neo4j/driver/internal/driver_factory.rb".freeze, "ruby/neo4j/driver/internal/handlers/begin_tx_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/channel_releasing_reset_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/commit_tx_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/hello_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/init_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/legacy_pull_all_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/no_op_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/ping_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/pull_handlers.rb".freeze, "ruby/neo4j/driver/internal/handlers/pulln/auto_pull_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/pulln/basic_pull_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/pulln/fetch_size_util.rb".freeze, "ruby/neo4j/driver/internal/handlers/reset_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/rollback_tx_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/route_message_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/routing_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/run_response_handler.rb".freeze, "ruby/neo4j/driver/internal/handlers/session_pull_response_completion_listener.rb".freeze, "ruby/neo4j/driver/internal/handlers/transaction_pull_response_completion_listener.rb".freeze, "ruby/neo4j/driver/internal/impersonation_util.rb".freeze, "ruby/neo4j/driver/internal/internal_bookmark.rb".freeze, "ruby/neo4j/driver/internal/internal_database_name.rb".freeze, "ruby/neo4j/driver/internal/internal_driver.rb".freeze, "ruby/neo4j/driver/internal/internal_entity.rb".freeze, "ruby/neo4j/driver/internal/internal_iso_duration.rb".freeze, "ruby/neo4j/driver/internal/internal_node.rb".freeze, "ruby/neo4j/driver/internal/internal_pair.rb".freeze, "ruby/neo4j/driver/internal/internal_path.rb".freeze, "ruby/neo4j/driver/internal/internal_point2_d.rb".freeze, "ruby/neo4j/driver/internal/internal_point3_d.rb".freeze, "ruby/neo4j/driver/internal/internal_relationship.rb".freeze, "ruby/neo4j/driver/internal/internal_result.rb".freeze, "ruby/neo4j/driver/internal/internal_session.rb".freeze, "ruby/neo4j/driver/internal/internal_transaction.rb".freeze, "ruby/neo4j/driver/internal/logging/channel_activity_logger.rb".freeze, "ruby/neo4j/driver/internal/logging/channel_error_logger.rb".freeze, "ruby/neo4j/driver/internal/logging/console_logging.rb".freeze, "ruby/neo4j/driver/internal/logging/dev_null_logger.rb".freeze, "ruby/neo4j/driver/internal/logging/dev_null_logging.rb".freeze, "ruby/neo4j/driver/internal/logging/j_u_logger.rb".freeze, "ruby/neo4j/driver/internal/logging/j_u_logging.rb".freeze, "ruby/neo4j/driver/internal/logging/netty_logger.rb".freeze, "ruby/neo4j/driver/internal/logging/netty_logging.rb".freeze, "ruby/neo4j/driver/internal/logging/prefixed_logger.rb".freeze, "ruby/neo4j/driver/internal/logging/reformatted_logger.rb".freeze, "ruby/neo4j/driver/internal/logging/slf4j_logger.rb".freeze, "ruby/neo4j/driver/internal/logging/slf4j_logging.rb".freeze, "ruby/neo4j/driver/internal/messaging/abstract_message_writer.rb".freeze, "ruby/neo4j/driver/internal/messaging/bolt_protocol_version.rb".freeze, "ruby/neo4j/driver/internal/messaging/common/common_message_reader.rb".freeze, "ruby/neo4j/driver/internal/messaging/common/common_value_packer.rb".freeze, "ruby/neo4j/driver/internal/messaging/common/common_value_unpacker.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/begin_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/commit_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/discard_all_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/discard_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/goodbye_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/hello_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/init_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/pull_all_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/pull_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/reset_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/rollback_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/route_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/route_v44_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/run_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/encode/run_with_metadata_message_encoder.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/abstract_streaming_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/begin_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/commit_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/discard_all_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/discard_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/goodbye_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/hello_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/init_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/message_with_metadata.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/multi_database_util.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/pull_all_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/pull_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/reset_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/rollback_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/route_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/run_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/run_with_metadata_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/request/transaction_metadata_builder.rb".freeze, "ruby/neo4j/driver/internal/messaging/response/failure_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/response/ignored_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/response/record_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/response/success_message.rb".freeze, "ruby/neo4j/driver/internal/messaging/v3/bolt_protocol_v3.rb".freeze, "ruby/neo4j/driver/internal/messaging/v3/message_format_v3.rb".freeze, "ruby/neo4j/driver/internal/messaging/v3/message_writer_v3.rb".freeze, "ruby/neo4j/driver/internal/messaging/v4/bolt_protocol_v4.rb".freeze, "ruby/neo4j/driver/internal/messaging/v4/message_format_v4.rb".freeze, "ruby/neo4j/driver/internal/messaging/v4/message_writer_v4.rb".freeze, "ruby/neo4j/driver/internal/messaging/v41/bolt_protocol_v41.rb".freeze, "ruby/neo4j/driver/internal/messaging/v42/bolt_protocol_v42.rb".freeze, "ruby/neo4j/driver/internal/messaging/v43/bolt_protocol_v43.rb".freeze, "ruby/neo4j/driver/internal/messaging/v43/message_format_v43.rb".freeze, "ruby/neo4j/driver/internal/messaging/v43/message_writer_v43.rb".freeze, "ruby/neo4j/driver/internal/messaging/v44/bolt_protocol_v44.rb".freeze, "ruby/neo4j/driver/internal/messaging/v44/message_format_v44.rb".freeze, "ruby/neo4j/driver/internal/messaging/v44/message_writer_v44.rb".freeze, "ruby/neo4j/driver/internal/metrics/internal_abstract_metrics.rb".freeze, "ruby/neo4j/driver/internal/metrics/internal_connection_pool_metrics.rb".freeze, "ruby/neo4j/driver/internal/metrics/internal_metrics.rb".freeze, "ruby/neo4j/driver/internal/metrics/internal_metrics_provider.rb".freeze, "ruby/neo4j/driver/internal/metrics/time_recorder_listener_event.rb".freeze, "ruby/neo4j/driver/internal/packstream/byte_array_incompatible_packer.rb".freeze, "ruby/neo4j/driver/internal/packstream/pack_stream.rb".freeze, "ruby/neo4j/driver/internal/packstream/pack_type.rb".freeze, "ruby/neo4j/driver/internal/reactive/internal_rx_result.rb".freeze, "ruby/neo4j/driver/internal/reactive/internal_rx_session.rb".freeze, "ruby/neo4j/driver/internal/reactive/internal_rx_transaction.rb".freeze, "ruby/neo4j/driver/internal/reactive/rx_utils.rb".freeze, "ruby/neo4j/driver/internal/retry/exponential_backoff_retry_logic.rb".freeze, "ruby/neo4j/driver/internal/revocation_strategy.rb".freeze, "ruby/neo4j/driver/internal/scheme.rb".freeze, "ruby/neo4j/driver/internal/security/internal_auth_token.rb".freeze, "ruby/neo4j/driver/internal/security/security_plan_impl.rb".freeze, "ruby/neo4j/driver/internal/security_setting.rb".freeze, "ruby/neo4j/driver/internal/session_factory_impl.rb".freeze, "ruby/neo4j/driver/internal/spi/connection.rb".freeze, "ruby/neo4j/driver/internal/spi/connection_pool.rb".freeze, "ruby/neo4j/driver/internal/spi/response_handler.rb".freeze, "ruby/neo4j/driver/internal/summary/internal_database_info.rb".freeze, "ruby/neo4j/driver/internal/summary/internal_input_position.rb".freeze, "ruby/neo4j/driver/internal/summary/internal_notification.rb".freeze, "ruby/neo4j/driver/internal/summary/internal_plan.rb".freeze, "ruby/neo4j/driver/internal/summary/internal_profiled_plan.rb".freeze, "ruby/neo4j/driver/internal/summary/internal_result_summary.rb".freeze, "ruby/neo4j/driver/internal/summary/internal_server_info.rb".freeze, "ruby/neo4j/driver/internal/summary/internal_summary_counters.rb".freeze, "ruby/neo4j/driver/internal/svm/netty_substitutions.rb".freeze, "ruby/neo4j/driver/internal/svm/z_lib_substitutions.rb".freeze, "ruby/neo4j/driver/internal/util/certificate_tool.rb".freeze, "ruby/neo4j/driver/internal/util/clock.rb".freeze, "ruby/neo4j/driver/internal/util/error_util.rb".freeze, "ruby/neo4j/driver/internal/util/extract.rb".freeze, "ruby/neo4j/driver/internal/util/format.rb".freeze, "ruby/neo4j/driver/internal/util/futures.rb".freeze, "ruby/neo4j/driver/internal/util/iterables.rb".freeze, "ruby/neo4j/driver/internal/util/lock_util.rb".freeze, "ruby/neo4j/driver/internal/util/metadata_extractor.rb".freeze, "ruby/neo4j/driver/internal/util/preconditions.rb".freeze, "ruby/neo4j/driver/internal/util/query_keys.rb".freeze, "ruby/neo4j/driver/internal/util/server_version.rb".freeze, "ruby/neo4j/driver/logger.rb".freeze, "ruby/neo4j/driver/logging.rb".freeze, "ruby/neo4j/driver/net/server_address1.rb".freeze, "ruby/neo4j/driver/records.rb".freeze, "ruby/neo4j/driver/session_config.rb".freeze, "ruby/neo4j/driver/summary/query_type1.rb".freeze]
  s.homepage = "https://github.com/neo4jrb/neo4j-ruby-driver".freeze
  s.licenses = ["MIT".freeze]
  s.rdoc_options = ["--main".freeze, "README.md".freeze]
  s.required_ruby_version = Gem::Requirement.new(">= 2.6".freeze)
  s.requirements = ["jar org.neo4j.driver, neo4j-java-driver, 4.4.2".freeze]
  s.rubygems_version = "3.2.29".freeze
  s.summary = "".freeze

  if s.respond_to? :specification_version then
    s.specification_version = 4
  end

  if s.respond_to? :add_runtime_dependency then
    s.add_runtime_dependency(%q<activesupport>.freeze, [">= 0"])
    s.add_runtime_dependency(%q<zeitwerk>.freeze, [">= 2.1.10"])
    s.add_runtime_dependency(%q<jar-dependencies>.freeze, [">= 0"])
    s.add_development_dependency(%q<ffaker>.freeze, [">= 0"])
    s.add_development_dependency(%q<hoe>.freeze, [">= 0"])
    s.add_development_dependency(%q<hoe-bundler>.freeze, [">= 0"])
    s.add_development_dependency(%q<hoe-gemspec>.freeze, [">= 0"])
    s.add_development_dependency(%q<parallel>.freeze, [">= 0"])
    s.add_development_dependency(%q<rake>.freeze, [">= 0"])
    s.add_development_dependency(%q<rspec-its>.freeze, [">= 0"])
    s.add_development_dependency(%q<rspec-mocks>.freeze, [">= 0"])
    s.add_development_dependency(%q<ruby-maven>.freeze, [">= 0"])
    s.add_development_dependency(%q<rdoc>.freeze, [">= 4.0", "< 7"])
  else
    s.add_dependency(%q<activesupport>.freeze, [">= 0"])
    s.add_dependency(%q<zeitwerk>.freeze, [">= 2.1.10"])
    s.add_dependency(%q<jar-dependencies>.freeze, [">= 0"])
    s.add_dependency(%q<ffaker>.freeze, [">= 0"])
    s.add_dependency(%q<hoe>.freeze, [">= 0"])
    s.add_dependency(%q<hoe-bundler>.freeze, [">= 0"])
    s.add_dependency(%q<hoe-gemspec>.freeze, [">= 0"])
    s.add_dependency(%q<parallel>.freeze, [">= 0"])
    s.add_dependency(%q<rake>.freeze, [">= 0"])
    s.add_dependency(%q<rspec-its>.freeze, [">= 0"])
    s.add_dependency(%q<rspec-mocks>.freeze, [">= 0"])
    s.add_dependency(%q<ruby-maven>.freeze, [">= 0"])
    s.add_dependency(%q<rdoc>.freeze, [">= 4.0", "< 7"])
  end
end
```

**Comment by headius** [comment]
Hey, sorry for delay in looking at this!

I'm not sure exactly how to reproduce this, but does deleting the comment and whitespace line out of `gemspec_pom.rb` help?

If you could push a small repo with a simple script to reproduce, it would help us figure out what regressed in 9.3.3 so we can fix it for 9.3.4.

**Comment by klobuczek** [comment]
@hedius this error is intermittent. It happens only in github actions. On average 5% of runs fail that way, but since we have build matrix of 10 it takes an average of 2 runs to have them all pass.
This error: 
```
Error: Non-parseable POM /home/runner/.rubies/jruby-9.3.3.0/lib/ruby/stdlib/jars/gemspec_pom.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:1) @ line 1, column 1 -> [Help 2]
```
is a strong indication that a ruby file is tried to be parsed as a pom.xml. (note the reference to a "start tag"). 
I could imagine the issue is in jar-dependecies or ruby-maven.
I have created a minimal project that exposes a similar problem on jruby-9.2.20.1 and jruby-9.3.3.0 but not on jruby-9.2.17.0: https://github.com/klobuczek/pom_bug and the workflow execution: https://github.com/klobuczek/pom_bug/actions/runs/1814997945

**Comment by mohits** [comment]
@klobuczek - that's the wrong headius in your message :)

**Comment by Hedius** [comment]
> @klobuczek - that's the wrong headius in your message :)

Hmmmm :smile:

**Comment by klobuczek** [comment]
sorry @headius and @Hedius

**Comment by bensheldon** [comment]
I started seeing something very similar to this on JRuby 9.4. It just started happening next week to all my JRuby builds 😬

[Example build](https://github.com/bensheldon/good_job/actions/runs/9374815474/job/25812413549#step:11:907) with these lines:


```
Installing actionmailer 7.1.3.4
Fetching actionmailbox 7.1.3.4
Fetching actiontext 7.1.3.4
Installing actionmailbox 7.1.3.4
Installing actiontext 7.1.3.4
Error: ] Some problems were encountered while processing the POMs:
[FATAL] Non-parseable POM /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/jars/gemspec_pom.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:2)  @ line 1, column 2
 @ 
Error:  The build could not read 1 project -> [Help 1]
Error:    
Error:    The project  (/home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/jars/gemspec_pom.rb) has 1 error
Error:      Non-parseable POM /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/jars/gemspec_pom.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:2)  @ line 1, column 2 -> [Help 2]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
Error:  [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/ModelParseException
--- ERROR REPORT TEMPLATE -------------------------------------------------------
```
```
Errno::ENOENT: No such file or directory - /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/psych-5.1.2-java/deps.lst
  org/jruby/RubyIO.java:1278:in `sysopen'
  org/jruby/RubyIO.java:4264:in `read'
  /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/jars/installer.rb:69:in `load_from_maven'
  /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/jars/installer.rb:235:in `install_dependencies'
  /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/jars/installer.rb:224:in `do_install'
  /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/jars/installer.rb:170:in `vendor_jars'
  /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/jars/post_install_hook.rb:28:in `block in <main>'
  /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/rubygems/installer.rb:383:in `block in run_post_install_hooks'
  org/jruby/RubyArray.java:1981:in `each'
  /home/runner/.rubies/jruby-9.4.7.0/lib/ruby/stdlib/rubygems/installer.rb:382:in `run_post_install_hooks'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/rubygems_gem_installer.rb:43:in `install'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/source/rubygems.rb:207:in `install'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/installer/gem_installer.rb:54:in `install'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/installer/parallel_installer.rb:186:in `do_install'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/installer/parallel_installer.rb:177:in `block in worker_pool'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/worker.rb:62:in `apply_func'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/worker.rb:57:in `block in process_queue'
  org/jruby/RubyKernel.java:1722:in `loop'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/worker.rb:54:in `process_queue'
  /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/gems/bundler-2.3.26/lib/bundler/worker.rb:91:in `block in create_threads'

```

## Environment

```
Bundler       2.3.26
  Platforms   ruby, universal-java-11
Ruby          3.1.4p0 (2024-04-29 revision 597ff08ac1e0d7f05f8f6cc836648725756dd7db) [universal-java-11]
  Full Path   /home/runner/.rubies/jruby-9.4.7.0/bin/jruby
  Config Dir  /home/runner/.rubies/jruby-9.4.7.0/etc
RubyGems      3.5.11
  Gem Home    /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0
  Gem Path    /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0
  User Home   /home/runner
  User Path   /home/runner/.local/share/gem/jruby/3.1.0
  Bin Dir     /home/runner/work/good_job/good_job/vendor/bundle/jruby/3.1.0/bin
OpenSSL       
  Compiled    JRuby-OpenSSL 0.14.5
  Loaded      JRuby-OpenSSL 0.14.5
  Cert File   /usr/lib/jvm/temurin-11-jdk-amd64/lib/security/cacerts
  Cert Dir    /etc/ssl/certs
Tools         
  Git         2.45.1
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

**Comment by unflxw** [comment]
We are also [encountering a similar issue](https://appsignal.semaphoreci.com/jobs/33115ab3-dfab-4553-aaff-ff29aa2e8b3c) to the one @bensheldon is encountering:

```
Fetching actionmailbox 7.1.3.4
Fetching actiontext 7.1.3.4
Installing actionmailbox 7.1.3.4
Installing actiontext 7.1.3.4
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-parseable POM /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/jars/gemspec_pom.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:2)  @ line 1, column 2
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project  (/home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/jars/gemspec_pom.rb) has 1 error
[ERROR]     Non-parseable POM /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/jars/gemspec_pom.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:2)  @ line 1, column 2 -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/ModelParseException
```

```
Errno::ENOENT: No such file or directory - /home/semaphore/appsignal-ruby/.bundle/jruby/3.1.0/gems/psych-5.1.2-java/deps.lst
  org/jruby/RubyIO.java:1255:in `sysopen'
          org/jruby/RubyIO.java:4012:in `read'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/jars/installer.rb:69:in `load_from_maven'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/jars/installer.rb:235:in `install_dependencies'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/jars/installer.rb:224:in `do_install'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/jars/installer.rb:170:in `vendor_jars'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/jars/post_install_hook.rb:28:in `block in <main>'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/rubygems/installer.rb:383:in `block in run_post_install_hooks'
          org/jruby/RubyArray.java:1988:in `each'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/rubygems/installer.rb:382:in `run_post_install_hooks'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/rubygems_gem_installer.rb:43:in `install'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/source/rubygems.rb:206:in `install'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/installer/gem_installer.rb:54:in `install'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/installer/parallel_installer.rb:132:in `do_install'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/installer/parallel_installer.rb:123:in `block in worker_pool'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/worker.rb:62:in `apply_func'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/worker.rb:57:in `block in process_queue'
          org/jruby/RubyKernel.java:1586:in `loop'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/worker.rb:54:in `process_queue'
          /home/semaphore/.rbenv/versions/jruby-9.4.1.0/lib/ruby/stdlib/bundler/worker.rb:90:in `block in create_threads'
```

Did some digging into @bensheldon's CI setup to try and spot similarities and differences between ours and theirs:
- Different JRuby versions (issue happens with `9.4.7.0` and some `9.3` on theirs, and on `9.4.1.0` and `9.4.5.0` on ours)
- Different CI providers (GitHub Actions on theirs, Semaphore on ours)
- Ours does not have a lockfile, theirs does
- Both installing from a gemspec
- Both installing to a `BUNDLE_PATH`
- Both have `rails` as dependency (might be relevant since the error occurs after installing `actiontext`)

**Comment by headius** [comment]
I'd like to look into this issue this week. Can either of you produce something I can use to reproduce the errors?

I don't know why 9.4.1.0 or 9.4.7.0 would suddenly start doing this unless one of the upstream gems made changes. In both cases it looks a bit like the psych gem may be involved but that version was released over six months ago.

Did either of you update any gems to get to this issue?

**Comment by bensheldon** [comment]
Unfortunately I delete the Gemfile.lock when building in CI (so dependencies are not locked per se) but I do log the generated lockfile. These were the gems that were updated between when it I had a successful build and when it started failing:

```diff
--- before.txt	2024-06-17 11:56:22
+++ after.txt	2024-06-17 11:56:11
@@ -16,35 +16,35 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    actioncable (7.1.3.3)
-      actionpack (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    actioncable (7.1.3.4)
+      actionpack (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       nio4r (~> 2.0)
       websocket-driver (>= 0.6.1)
       zeitwerk (~> 2.6)
-    actionmailbox (7.1.3.3)
-      actionpack (= 7.1.3.3)
-      activejob (= 7.1.3.3)
-      activerecord (= 7.1.3.3)
-      activestorage (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    actionmailbox (7.1.3.4)
+      actionpack (= 7.1.3.4)
+      activejob (= 7.1.3.4)
+      activerecord (= 7.1.3.4)
+      activestorage (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       mail (>= 2.7.1)
       net-imap
       net-pop
       net-smtp
-    actionmailer (7.1.3.3)
-      actionpack (= 7.1.3.3)
-      actionview (= 7.1.3.3)
-      activejob (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    actionmailer (7.1.3.4)
+      actionpack (= 7.1.3.4)
+      actionview (= 7.1.3.4)
+      activejob (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       mail (~> 2.5, >= 2.5.4)
       net-imap
       net-pop
       net-smtp
       rails-dom-testing (~> 2.2)
-    actionpack (7.1.3.3)
-      actionview (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    actionpack (7.1.3.4)
+      actionview (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       nokogiri (>= 1.8.5)
       racc
       rack (>= 2.2.4)
@@ -52,40 +52,40 @@
       rack-test (>= 0.6.3)
       rails-dom-testing (~> 2.2)
       rails-html-sanitizer (~> 1.6)
-    actiontext (7.1.3.3)
-      actionpack (= 7.1.3.3)
-      activerecord (= 7.1.3.3)
-      activestorage (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    actiontext (7.1.3.4)
+      actionpack (= 7.1.3.4)
+      activerecord (= 7.1.3.4)
+      activestorage (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       globalid (>= 0.6.0)
       nokogiri (>= 1.8.5)
-    actionview (7.1.3.3)
-      activesupport (= 7.1.3.3)
+    actionview (7.1.3.4)
+      activesupport (= 7.1.3.4)
       builder (~> 3.1)
       erubi (~> 1.11)
       rails-dom-testing (~> 2.2)
       rails-html-sanitizer (~> 1.6)
-    activejob (7.1.3.3)
-      activesupport (= 7.1.3.3)
+    activejob (7.1.3.4)
+      activesupport (= 7.1.3.4)
       globalid (>= 0.3.6)
-    activemodel (7.1.3.3)
-      activesupport (= 7.1.3.3)
-    activerecord (7.1.3.3)
-      activemodel (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    activemodel (7.1.3.4)
+      activesupport (= 7.1.3.4)
+    activerecord (7.1.3.4)
+      activemodel (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       timeout (>= 0.4.0)
     activerecord-jdbc-adapter (70.2-java)
       activerecord (~> 7.0)
     activerecord-jdbcpostgresql-adapter (70.2-java)
       activerecord-jdbc-adapter (= 70.2)
       jdbc-postgres (>= 9.4, < 43)
-    activestorage (7.1.3.3)
-      actionpack (= 7.1.3.3)
-      activejob (= 7.1.3.3)
-      activerecord (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    activestorage (7.1.3.4)
+      actionpack (= 7.1.3.4)
+      activejob (= 7.1.3.4)
+      activerecord (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       marcel (~> 1.0)
-    activesupport (7.1.3.3)
+    activesupport (7.1.3.4)
       base64
       bigdecimal
       concurrent-ruby (~> 1.0, >= 1.0.2)
@@ -103,7 +103,7 @@
       thor (>= 0.14.0)
     base64 (0.2.0)
     bigdecimal (3.1.8-java)
-    builder (3.2.4)
+    builder (3.3.0)
     capybara (3.40.0)
       addressable
       matrix
@@ -113,13 +113,13 @@
       rack-test (>= 0.6.3)
       regexp_parser (>= 1.5, < 3.0)
       xpath (~> 3.2)
-    concurrent-ruby (1.2.3)
+    concurrent-ruby (1.3.3)
     connection_pool (2.4.1)
     crass (1.0.6)
     date (3.3.4-java)
     diff-lcs (1.5.1)
     drb (2.2.1)
-    erubi (1.12.0)
+    erubi (1.13.0)
     et-orbi (1.2.11)
       tzinfo
     fugit (1.11.0)
@@ -152,7 +152,7 @@
     mini_mime (1.1.5)
     minitest (5.23.1)
     mutex_m (0.2.0)
-    net-imap (0.4.11)
+    net-imap (0.4.12)
       date
       net-protocol
     net-pop (0.1.2)
@@ -162,7 +162,7 @@
     net-smtp (0.5.0)
       net-protocol
     nio4r (2.7.3-java)
-    nokogiri (1.16.5-java)
+    nokogiri (1.16.6-java)
       racc (~> 1.4)
     psych (5.1.2-java)
       jar-dependencies (>= 0.1.7)
@@ -179,20 +179,20 @@
     rackup (1.0.0)
       rack (< 3)
       webrick
-    rails (7.1.3.3)
-      actioncable (= 7.1.3.3)
-      actionmailbox (= 7.1.3.3)
-      actionmailer (= 7.1.3.3)
-      actionpack (= 7.1.3.3)
-      actiontext (= 7.1.3.3)
-      actionview (= 7.1.3.3)
-      activejob (= 7.1.3.3)
-      activemodel (= 7.1.3.3)
-      activerecord (= 7.1.3.3)
-      activestorage (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    rails (7.1.3.4)
+      actioncable (= 7.1.3.4)
+      actionmailbox (= 7.1.3.4)
+      actionmailer (= 7.1.3.4)
+      actionpack (= 7.1.3.4)
+      actiontext (= 7.1.3.4)
+      actionview (= 7.1.3.4)
+      activejob (= 7.1.3.4)
+      activemodel (= 7.1.3.4)
+      activerecord (= 7.1.3.4)
+      activestorage (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       bundler (>= 1.15.0)
-      railties (= 7.1.3.3)
+      railties (= 7.1.3.4)
     rails-dom-testing (2.2.0)
       activesupport (>= 5.0.0)
       minitest
@@ -200,9 +200,9 @@
     rails-html-sanitizer (1.6.0)
       loofah (~> 2.21)
       nokogiri (~> 1.14)
-    railties (7.1.3.3)
-      actionpack (= 7.1.3.3)
-      activesupport (= 7.1.3.3)
+    railties (7.1.3.4)
+      actionpack (= 7.1.3.4)
+      activesupport (= 7.1.3.4)
       irb
       rackup (>= 1.0.0)
       rake (>= 12.2)
@@ -212,13 +212,13 @@
     rdoc (6.7.0)
       psych (>= 4.0.0)
     regexp_parser (2.9.2)
-    reline (0.5.7)
+    reline (0.5.9)
       io-console (~> 0.5)
-    rexml (3.2.8)
-      strscan (>= 3.0.9)
+    rexml (3.3.0)
+      strscan
     rspec-core (3.13.0)
       rspec-support (~> 3.13.0)
-    rspec-expectations (3.13.0)
+    rspec-expectations (3.13.1)
       diff-lcs (>= 1.2.0, < 2.0)
       rspec-support (~> 3.13.0)
     rspec-mocks (3.13.1)
@@ -241,7 +241,7 @@
       websocket (~> 1.0)
     strscan (3.1.0-java)
     thor (1.3.1)
-    timecop (0.9.8)
+    timecop (0.9.9)
     timeout (0.4.1)
     tzinfo (2.0.6)
       concurrent-ruby (~> 1.0)
@@ -255,7 +255,7 @@
     yard (0.9.36)
     yard-activesupport-concern (0.0.1)
       yard (>= 0.8)
-    zeitwerk (2.6.14)
+    zeitwerk (2.6.15)

 PLATFORMS
   universal-java-11
```

**Comment by unflxw** [comment]
> In both cases it looks a bit like the psych gem may be involved but that version was released over six months ago.

I thought the same, but I have been able to reproduce the issue with several `psych` versions (`5.0.1`, `5.1.0`, `5.1.1.1`) -- the issue seems to take place irrespective of the `psych` version and the JRuby version.

> I delete the Gemfile.lock when building in CI [...] but I do log the generated lockfile

Smart decision! Unfortunately we don't log our installed dependencies (hindsight is 20/20) but we do have two builds nine hours apart ([earlier working](https://appsignal.semaphoreci.com/jobs/9bb9d03e-f3e8-48d6-93b3-ca03a44cc63b#L748), [later failing](https://appsignal.semaphoreci.com/jobs/94ed6b1c-6a13-46d5-8498-3874b56ab64e#L854)) -- since gems are cached between builds, the list of installed packages in the working one is incomplete, but it does narrow down the list somewhat -- specifically it exonerates the `action*` gem version bumps (though the issue might still lie in one of their transitive dependencies!)

**Comment by headius** [comment]
If my reading of the logic in jar-dependencies is right, the missing file is supposed to have been generated by the maven build that appears to fail.

The following logic is the source of that missing filename:

https://github.com/mkristian/jar-dependencies/blob/b4e80f0d29cf5030dde566c22b4d8a32e3610681/lib/jars/installer.rb#L210-L219

That filename is passed along to a maven run to be generated, by using maven to resolve all jar-based dependencies and then output this `deps.lst` file. The file is then used to vendor the gems into their appropriate resting place:

https://github.com/mkristian/jar-dependencies/blob/b4e80f0d29cf5030dde566c22b4d8a32e3610681/lib/jars/maven_exec.rb#L67-L83

I believe the failure to read this file is just a symptom of the maven execution failing above, so I'm going to focus on what's happening there for now.

**Comment by headius** [comment]
One suspect: ruby-maven-libs was recently updated (early May) to a newer version of Maven (3.9.x):

https://rubygems.org/gems/ruby-maven-libs

It was previously updated in March (3.8.x) but prior to that there were no updates since 2016. @Kinju011 filed this issue in February so that throws some cold water on this theory.

**Comment by headius** [comment]
There were also some updates of polyglot-ruby toward the end of 2023:

https://central.sonatype.com/artifact/io.takari.polyglot/polyglot-ruby/versions

Polyglot is used to write Maven pom files in other languages, such as the `gemspec_pom.rb` that triggers the Maven errors in y'alls build output.

**Comment by headius** [comment]
@mkristian I filed takari/polyglot-maven#312 in case it's something broken on the polyglot side (I don't see why it would generate a pom.xml with Ruby comments in it) but perhaps you have other thoughts on what could be causing this for @bensheldon and @unflxw?

**Comment by unflxw** [comment]
Thank you for looking into this @headius!

I will try to produce a reproducible test case to aid in figuring this out -- if I can produce it, I'll post it both here and in the `polyglot-maven` issue. My gut feeling at the moment, regarding reproducing it:
- It most likely **is related to gem dependencies**, since both of our projects build without locks, and we started experiencing this issue at the same time, a couple of days ago.
- It likely **is not related to the underlying JRuby or Java version**. We build from CI environments that usually wouldn't update JRuby versions on the fly, and Semaphore in particular does not update dependencies often.
- Specifically, I think, it **is likely to be due to a transitive dependency of Rails 7.1**, since it's the dependency we have in common, and the issue does not take place in our (also lock-less) Rails 7.0 test setup.
- I also think **it is likely to be related to building _from a gemspec_**. If installing Rails alone from a Gemfile could reproduce this, more people would be running into this.

**Comment by unflxw** [comment]
Okay, I have found a minimal reproduction test case, and it indicates the issue lies with recent versions of RubyGems or Bundler.

(Our CI does `gem update --system` and `gem update bundler` on each run)

From inside a JRuby environment, such as `docker run -it jruby:latest /bin/bash`:

```
gem update --system
gem update bundler
echo -e "source 'https://rubygems.org/'\ngem 'rails'" > Gemfile
bundle install
```

**Comment by headius** [comment]
Great repro! I will investigate.

**Comment by headius** [comment]
So a quick workaround seems to be installing `jar-dependencies` and `ruby-maven` before running bundle install. I suspect there's some interaction happening when we install those during the installation of psych that's screwing up the fetching of jar dependencies.

**Comment by headius** [comment]
Another discovery: if I hack the jar-dependencies installer to use ruby-maven 3.3.9 (rather than the latest ones we pushed in the past few months), it installs successfully. It seems like there's a problem related to the new maven bits in the newer ruby-maven gems.

**Comment by headius** [comment]
@deivid-rodriguez @jsvd @ccutrer Hey folks, I need more eyes on this.

Something about the various updates that happened in jar-dependencies, ruby-maven, and ruby-libs over the past couple of years have caused it to sometimes fail to recognize the jar-dependencies `gemspec_pom.rb` as a Polyglot Ruby pom file, resulting in the error seen here.

The original report was in early 2022, which is baffling because many of the updates I suspected (updating Maven version, updating Polyglot used by ruby-maven, etc) happened after that report. Several sets of changes from @deivid-rodriguez happened around that time but I did not see any that looked like they'd cause this.

The repro by @unflxw works well. The problem seems to be that when ruby-maven initializes it's not finding its own `.mvn/extensions.xml` which should be in the installed gem. ruby-maven is installed during a post-install hook in the affected environments, when jar-dependencies tries to install dependent jars. It asks for latest ruby-maven and ruby-maven-libs to be installed and then proceeds to tell ruby-maven to run gemspec_pom.rb. If I force jar-dependencies to install ruby-maven-libs 3.3.9 instead of latest, it appears to succeed.

I will be in office on Matrix on and off over the next 24h but eager to get some help investigating this.

**Comment by headius** [comment]
Oh worth noting that for @unflxw example, if you don't update the RubyGems version, it works ok. Has something changed recently that might make it problematic to install more gems during a post-install hook?

**Comment by headius** [comment]
I may have a solid clue here...

It seems like Maven ignores the local .mvn directory when:

1. You are specifying a -f file in a different directory
2. You are at the root path (`/`)

Here's an example running inside a jruby docker container:

```
root@3d97b6fce93a:/# cat .mvn/extensions.xml 
<?xml version="1.0" encoding="UTF-8"?>
<extensions>
	<!--<extension>
    <groupId>org.torquebox.mojo</groupId>
    <artifactId>mavengem-wagon</artifactId>
    <version>1.0.3</version>
	</extension>-->
	<extension>
    <groupId>io.takari.polyglot</groupId>
    <artifactId>polyglot-ruby</artifactId>
    <version>0.7.0</version>
  </extension>
</extensions>
root@3d97b6fce93a:/# vi .mvn/extensions.xml 
root@3d97b6fce93a:/# cat .mvn/extensions.xml 
<?xml version="1.0" encoding="UTF-8"?>
<extensions>
  <extension>
    <groupId>io.takari.polyglot</groupId>
    <artifactId>polyglot-ruby</artifactId>
    <version>0.7.0</version>
  </extension>
</extensions>
root@3d97b6fce93a:/# cat blah.rb
#
root@3d97b6fce93a:/# cat blah/blah.rb
#
root@3d97b6fce93a:/# mvn -f blah.rb clean
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'artifactId' with value '/' does not match a valid id pattern. @ 
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project no_group_id_given:/:0.0.0 (/blah.rb) has 1 error
[ERROR]     'artifactId' with value '/' does not match a valid id pattern.
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
root@3d97b6fce93a:/# mvn -f blah/blah.rb clean
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-parseable POM /blah/blah.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:1)  @ line 1, column 1
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project  (/blah/blah.rb) has 1 error
[ERROR]     Non-parseable POM /blah/blah.rb: only whitespace content allowed before start tag and not # (position: START_DOCUMENT seen #... @1:1)  @ line 1, column 1 -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/ModelParseException
```

The first error is from running against `blah.rb` in the current (root) directory. Notice that it has trouble determining the artifact name because it's at the root path, but otherwise it did recognize the file as a Ruby-based pom file.

The second error occurs when referencing `blah/blah.rb` as the build file. The extensions are not loaded and it is not recognized as a Ruby-based pom file.

Now if we move everything into a non-root dir:

```
root@3d97b6fce93a:/# mv .mvn blah* tmp
root@3d97b6fce93a:/# cd tmp
root@3d97b6fce93a:/tmp# mvn -f blah.rb clean
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< no_group_id_given:tmp >------------------------
[INFO] Building tmp 0.0.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ tmp ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.454 s
[INFO] Finished at: 2024-06-26T06:53:01Z
[INFO] ------------------------------------------------------------------------
root@3d97b6fce93a:/tmp# mvn -f blah/blah.rb clean
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< no_group_id_given:blah >-----------------------
[INFO] Building blah 0.0.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ blah ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.381 s
[INFO] Finished at: 2024-06-26T06:53:06Z
[INFO] ------------------------------------------------------------------------
```

Both commands succeed.

I'm thinking this is a Maven bug with extension loading when executing from a root path. This example was running with Maven 3.6.3.

**Comment by headius** [comment]
This is probably a double-edged bug.

I filed an issue with Maven here: https://issues.apache.org/jira/browse/MNG-8165

Basically I think the problem is that when they "fixed" Maven to look in the target `-f` file's directory for `.mvn`, they broke any ability to look at the root dir for `.mvn`. That effectively kills the way jar-dependencies tries to run its Polyglot Ruby pom file when running at root. It searches the target file's dir and parent dirs, but never gets to root.

But there's also a bug in ruby-maven in that it's depending on old Maven behavior that looked in the current directory for `.mvn` even if you specified a different pom path with `-f`. We probably will get a quicker and more reliable result by fixing ruby-maven to more robustly load its extensions when running Maven commands, rather than hoping that the old behavior will come back or the new behavior will search to root (which wouldn't fix cases where ruby-maven, JRuby, and CWD are in very different directory hierarchies).

**Comment by headius** [comment]
I believe I have a workaround: install an `extensions.xml` file into the JRuby distribution under `lib/ruby/stdlib/jars/.mvn`.

```
root@3d97b6fce93a:/tmp# cp -R .mvn /opt/jruby/lib/ruby/stdlib/jars/
root@3d97b6fce93a:/tmp# cd ..
root@3d97b6fce93a:/# gem uninstall ruby-maven-libs psych
Gem psych-5.1.1.1-java cannot be uninstalled because it is a default gem
Successfully uninstalled psych-5.1.2-java
Successfully uninstalled ruby-maven-libs-3.8.9
root@3d97b6fce93a:/# bundle
Fetching gem metadata from https://rubygems.org/.......
Fetching psych 5.1.2 (java)
Installing psych 5.1.2 (java)
  jar dependencies for psych-5.1.2-java.gemspec . . .
Installing gem 'ruby-maven-libs' . . .

using maven for the first time results in maven
downloading all its default plugin and can take time.
as those plugins get cached on disk and further execution
of maven is much faster then the first time.

      org.snakeyaml:snakeyaml-engine:2.7:compile
Bundle complete! 1 Gemfile dependency, 3 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```

This may be the simplest fix, as well; any time you want to use ruby-maven to execute Polyglot Ruby pom file, that pom file's dir hierarchy must have the correct extension configuration.

**Comment by headius** [comment]
I've marked this for 9.4.8.0 but it may be something we want in 9.3.x as well. @enebo I think the fix in https://github.com/mkristian/jar-dependencies/pull/81 should be pretty benign, since it's just a config file for maven extensions. As noted there, though, it isn't a complete fix; ruby-maven will eventually need a patch to enable polyglot ruby all the time without juggling config files around.

**Comment by deivid-rodriguez** [comment]
@headius Did you figure this out in the end, or should I look into this as well?

**Comment by headius** [comment]
@deivid-rodriguez I think I have tracked it down to a behavioral change in Maven, but if I think you could help I'll let you know. Thank you!

**Comment by headius** [comment]
Two different fixes for this are in the pipeline:

* mkristian/jar-dependencies:81

This works around the bug in ruby-maven (https://github.com/jruby/ruby-maven/issues/9) by stuffing an appropriate `.mvn/extensions.xml` into the same dir as `gemspec_pom.rb` so ruby polyglot will work properly. It does not fix the real issue, which is that ruby-maven needs a more robust way to activate ruby polyglot for all users and all ways of invoking.

* https://github.com/jruby/ruby-maven/issues/9

This is the actual issue, and may be simplest to fix by including the ruby polyglot extension in ruby-maven-libs directly. I need help here, from @mkristian or anyone else familiar with Maven!

**Comment by headius** [comment]
Also Maven 3.9.9 will fix the smaller issue of not looking in the root directory for .mvn: https://issues.apache.org/jira/browse/MNG-8159

That fix would also work around this issue, but not fix the problem of ruby-maven depending on the current directory to enable extensions.

**Comment by headius** [comment]
I suspect this will be better fixed by doing something in ruby-maven, but either way it's too close to the release for us to patch it in 9.4.8.0.

**Comment by headius** [comment]
@mkristian It sounded like you had the ruby-maven changes in hand, but if there's anything I can do to help get that finished please let me know!

**Comment by headius** [comment]
As described [here](https://github.com/jruby/ruby-maven/issues/9#issuecomment-2220599492), polyglot will soon receive a patch to allow integrating it directly into a custom Maven distribution, and we will use that to build a ruby-maven gem that has ruby polyglot "always on". It will no longer depend on the location of .mvn/extensions.xml, avoiding this issue and fixing the problem without requiring a jar-dependencies or JRuby release.

We will want to remove the .mvn/extensions.xml file currently set up by jar-dependencies since it is no longer needed.

Currently waiting on the polyglot side so we can have something to start testing from ruby-maven up.

**Comment by headius** [comment]
Unfortunately, due to issues getting jar-dependencies updated (https://github.com/mkristian/jar-dependencies/issues/84) I think we need to punt this to the next release. jar-dependencies has a lot of old version dependencies and has not run integration tests in some time. I'd like to get it right when we update it.

**Comment by headius** [comment]
Should be fixed by the ruby-maven and jar-dependencies updates in #8477.
